test/socket.test.js

```diff
@@ -18,7 +18,10 @@ describe('socket', () => {
             socket = null;
         }
 
-        server.close(done);
+        server.close(err => {
+            // delay for freeing up port
```

Comment:
Isn't that what the callback is for?

I just delayed the callback a little more.